### PR TITLE
bug fix. remove cycle

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
remove buggy cycle
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Fixed issue in `CMakePresets.json` where Ninja presets were broken
+- Removed cycle in MAPL_Generic.F90
```
Comment (tclune) [inline CHANGELOG.md:13]: This is not a very useful note for the ChangeLog.   Users can look at "diff" to see the deleted line.   Here we should explain the why?  What was breaking what were the symptoms or consequences.

**Review by tclune — CHANGES_REQUESTED** [review]
See inline comment.

File: CHANGELOG.md (hunk 1 above)
Comment (weiyuan-jiang) [reply CHANGELOG.md:13]: That line was added by me. After thinking twice, I believe that "I" and "J" are from different sources and should not be cycled when they are the same.

**Comment by mathomp4** [comment]
I've put a blocker on my #1329 PR for now because this PR is here.

My question is: did you run GEOSgcm with this fix? I'd rather not merge it in if you haven't (the checkbox is not checked).

**Comment by weiyuan-jiang** [comment]
> 

Yes, I have tested GEOSgcm and it runs

File: CHANGELOG.md (hunk 1 above)
Comment (mathomp4) [reply CHANGELOG.md:13]: @weiyuan-jiang You can probably remove this line and then edit line 19:
```
- A small performance improvement. cycle => exit in MAPL_Generic.F90
```
to be whatever you think this PR plus the first on should reflect.

Comment (mathomp4) [reply CHANGELOG.md:13]: Yeah. I'll just remove this and add "skip changelog".

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**